File: juegos1000tres-frontend/src/comunicacion/Recibo.ts
import { Enviable, type EnviableConstructor } from "./Enviable.js";

export abstract class Recibo<PAYLOAD> {
  abstract traducirFormatoAEnviable<T extends Enviable>(
    payload: PAYLOAD,
    tipoEnviable: EnviableConstructor<T>
  ): T;

  abstract getTipoPayload(): string;
}

File: juegos1000tres-backend/src/main/java/com/juegos1000tres/juegos1000tres_backend/comunicacion/Recibo.java
package com.juegos1000tres.juegos1000tres_backend.comunicacion;

public interface Recibo<PAYLOAD> {

    <T extends Enviable> T traducirFormatoAEnviable(PAYLOAD payload, Class<T> tipoEnviable);

    Class<PAYLOAD> getClasePayload();
}